gensim/matutils.py

```diff
@@ -1328,10 +1328,23 @@ class MmReader(object):
     This allows us to process corpora which are larger than the available RAM.
 
     """
+    @contextmanager
+    def open_file(self, input):
+        # Generates 'with' like behaviour excepting closing the file object
+        mgr = utils.file_or_filename(self.input)
+        exc = False
+        try:
+            yield mgr
+        except StandardError:
+            exc = True
+            if not exit(mgr, *sys.exc_info()):
```

Comment:
I'm slightly worried about `exit`, how it works?

Reply:
In case of the exception , 'with' statement has the ability to silence errors. Here trying to close the file with the exception information if possible , otherwise raise  the exception.

Reply:
this is call `__exit__` ?

Reply:
Ya , it is __exit__() , I will correct it as in the finally block. Thanks.